content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1007 @@
+
+You can now iterate all permutations of N entities for a given query:
+
+```rust
+fn system(query: Query<&Player>) {
+    // iterates each possible combination of two entities exactly once
+    for [e1, e2] in query.iter_combinations() {
+    }
+
+    // iterates each possible combination of three entities exactly once
+    for [e1, e2, e3] in query.iter_combinations() {
+    }
+}
+```
+
+This is especially useful for things like "checking for entities for collisions with all other entities". There is also an `iter_combinations_mut` variant. Just be careful ... the time complexity of this grows quickly with large numbers of combinations. With great power comes great responsibility!
+
+The new `iter_combinations.rs` example illustrats how to use this new API to calculate gravity between objects in a "solar system":
+
+![iter_combinations](iter_combinations.png)
+
+### Optimized System Commands
+
+<div class="release-feature-authors">authors: @NathanSWard</div>
+
+System Commands got a nice performance boost by changing how command buffers are stored and reused:
+
+#### Entity Spawn Benchmark Duration (in microseconds, less is better)
+
+This benchmark spawns entities with a variety of component compositions to ensure we cover a variety of cases. Treat these numbers as realtives, not absolutes.
+
+![commands perf](commands_perf.svg)
+
+### System Param Lifetimes
+
+<div class="release-feature-authors">authors: @cart, @BoxyUwU, @TheRawMeatball</div>
+
+System and Query lifetimes were made more explicit by splitting out the `'system` and `'world`, lifetimes and using them explicitly where possible. This enables Rust to reason about ECS lifetimes more effectively, especially for read-only lifetimes. This was particularly important because it enabled the new Bevy Renderer to convince WGPU that ECS resources actually live for as long as the Render World.
+
+### Soundness / Correctness Improvements
+
+<div class="release-feature-authors">authors: @BoxyUwU, @TheRawMeatball, @Frizi, @thebluefish, @sapir, @bjorn3, @DJMcNab</div>
+
+Bevy ECS received a solid number of soundness and correctness improvements this release, alongside some unsafe code removals. Queries and internal storages like Tables and BlobVecs in particular had a number of improvements in these areas. And we have even more improvements in the works!
+
+### Hierarchy Convenience Functions
+
+<div class="release-feature-authors">authors: @TheRawMeatball, @jihiggins</div>
+
+```rust
+// Despawns all descendants of an entity (its children, its childrens' children, etc)
+commands.entity(e).despawn_descendants();
+
+// Removes the given children from the entity
+commands.entity(parent).remove_children(&[child1, child2]);
+```
+
+## UI
+
+### Overflow::Hidden
+
+<div class="release-feature-authors">authors: @Davier</div>
+
+UI now repects the flexbox `Overflow::Hidden` property. This can be used to cut off child content, which is useful when building things like scrollable lists:
```

Comment:
```suggestion
UI now respects the flexbox `Overflow::Hidden` property. This can be used to cut off child content, which is useful when building things like scrollable lists:
```